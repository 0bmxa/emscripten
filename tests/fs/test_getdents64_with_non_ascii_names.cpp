#include <fstream>
#include <cassert>
#include <dirent.h>
#include <sys/stat.h>
#include <fcntl.h>

void touch(const char* fileName)
{
    std::ofstream f(fileName, std::ios_base::out);
    assert(f.good());
}

int main()
{
    auto ret = mkdir("test_dir", 0777);
    assert(ret == 0);

    touch(u8"test_dir/абвг");

    auto fd = open("test_dir", O_RDONLY | O_DIRECTORY);
    assert(fd > 0);

    for (;;)
    {
        dirent d;
        auto nread = getdents(fd, &d, sizeof(d));
        assert(nread != -1);
        if (nread == 0)
        {
            break;
        }

        if (strcmp(d.d_name, ".") == 0 || strcmp(d.d_name, "..") == 0)
        {
            continue;
        }

        printf("d.d_name = %s\n", d.d_name);
    }
}
