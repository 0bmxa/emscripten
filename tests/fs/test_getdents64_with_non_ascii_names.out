d.d_name = абвг
